iothub/service/src/Twin/TwinsClient.cs

```diff
@@ -479,29 +455,12 @@ public virtual async Task<Twin> ReplaceAsync(string deviceId, Twin newTwin, bool
         /// <exception cref="OperationCanceledException">If the provided <paramref name="cancellationToken"/> has requested cancellation.</exception>
         public virtual async Task<Twin> ReplaceAsync(string deviceId, string newTwinJson, bool onlyIfUnchanged = false, CancellationToken cancellationToken = default)
         {
-            if (Logging.IsEnabled)
-                Logging.Enter(this, $"Replacing device twin on device: {deviceId}", nameof(ReplaceAsync));
-            try
-            {
-                Argument.AssertNotNullOrWhiteSpace(deviceId, nameof(deviceId));
-                Argument.AssertNotNullOrWhiteSpace(newTwinJson, nameof(newTwinJson));
-                cancellationToken.ThrowIfCancellationRequested();
+            Argument.AssertNotNullOrWhiteSpace(deviceId, nameof(deviceId));
+            Argument.AssertNotNullOrWhiteSpace(newTwinJson, nameof(newTwinJson));
 
-                // TODO: Do we need to deserialize Twin, only to serialize it again?
-                Twin twin = JsonConvert.DeserializeObject<Twin>(newTwinJson);
-                return await ReplaceAsync(deviceId, twin, onlyIfUnchanged, cancellationToken).ConfigureAwait(false);
-            }
-            catch (Exception ex)
-            {
-                if (Logging.IsEnabled)
-                    Logging.Error(this, $"{nameof(ReplaceAsync)} threw an exception: {ex}", nameof(ReplaceAsync));
-                throw;
-            }
-            finally
-            {
-                if (Logging.IsEnabled)
-                    Logging.Exit(this, $"Replacing device twin on device: {deviceId}", nameof(ReplaceAsync));
-            }
+            // TODO: Do we need to deserialize Twin, only to serialize it again?
```

Comment:
We can get rid of this TODO because the twin returned by the service has updated versions/etags/datetimes

Reply:
I think the question is about taking a string, turning it into a twin (to get the etag), and then when sending the request we serialize it back to a string.

And I think the answer to that is either the user must supply the etag as a parameter, or we must deserialize to get the etag.

Reply:
As discussed, these overloads are now gone and the user passes us a structured Twin object in all cases.